source/config.cpp

```diff
@@ -84,115 +84,107 @@ void Config::read(string const &file_name)
 	string line;
 
 	while( std::getline(f, line) ) {
-		if( line.size() ) {
-			if( line[0] == '#' ) continue;
+		if( !line.size() or line[0] == '#' ) continue;
 
-			if( line.back() == '\r' ) {
-				line.pop_back();
-				if( line.empty() ) continue;
+		if( line.back() == '\r' ) {
+			line.pop_back();
+			if( line.empty() ) continue;
+		}
+
+		if( !(line[0] == '+' or line[0] == '-') ) throw std::runtime_error("Invalid token at the begining of line in config file! Each line should begin with ether '+' or '-' or '#'! Line: " + line);
+		size_t space = line.find(' ');
+		if( space == string::npos )
+			throw std::runtime_error("Invalid line in config file! Each line must have token separated with space from object name. For example: '+function aaa::bb::my_function'. Line: " + line);
```

Comment:
since this does not fit on single line then we have to use brackets or it became error prone (say someone confuse this with Python and start add more lines after `if`, ie:

```
if( space == string::npos ) {
    throw std::runtime_error("Invalid line in config file! Each line must have token separated with space from object name. For example: '+function aaa::bb::my_function'. Line: " + line);
}
```

Reply:
ok, done